crates/matrix-sdk-indexeddb/src/crypto_store/mod.rs

```diff
@@ -827,13 +827,19 @@ impl_crypto_store! {
 
         let mut result = Vec::new();
         for _ in 0..limit {
-            result.push(self.deserialize_inbound_group_session(cursor.value())?);
+            result.push(cursor.value());
             if !cursor.continue_cursor()?.await? {
                 break;
             }
         }
 
         tx.await.into_result()?;
+
+        // deserialize and decrypt after the transaction is complete
+        let result = result.into_iter()
+            .filter_map(|v| self.deserialize_inbound_group_session(v).ok())
```

Comment:
Note for reviewer: There is a change compared to orginal because it won't error on deserialized, it will ignore now. But it's consistent with `get_inbound_group_sessions` for example

Reply:
added some logging